### PR TITLE
[new release] passage (0.1.0)

### PR DESCRIPTION
**Pull request by thatportugueseguy**
Passage - used to store and manage access to shared secrets

- Project page: <a href="https://github.com/ahrefs/passage">https://github.com/ahrefs/passage</a>

##### CHANGES:

- Open source passage

### CODE REVIEW

**Comment by thatportugueseguy** [comment]
CI is failing because some dependencies fail to build in windows. What are the guidelines for this situation, since it's not easy to update everything to ensure it builds on windows? thanks

**Comment by mseri** [comment]
That is not a problem, windows support can be added slowly. For the moment we have the windows CI as an indication and to help who is attempting to fix/port the packages

**Comment by mseri** [comment]
I have restarted the CI seems there seemed to be network issues

File: packages/passage/passage.0.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,52 @@
+opam-version: "2.0"
+synopsis: "Passage - used to store and manage access to shared secrets"
+description: "Passage - used to store and manage access to shared secrets"
+maintainer: ["Ahrefs Pte Ltd <github@ahrefs.com>"]
+authors: ["Ahrefs Pte Ltd <github@ahrefs.com>"]
+license: "MIT"
+homepage: "https://github.com/ahrefs/passage"
+bug-reports: "https://github.com/ahrefs/passage/issues"
+depends: [
+  "dune" {>= "3.9"}
```
Comment (mseri) [inline packages/passage/passage.0.1.0/opam:10]:
```suggestion
  "ocaml" {>= "4.14" & < "5.0"}
  "dune" {>= "3.9"}
```
Uses `String.starts_with` and `In_channel`, and relies on the parser:
```
#=== ERROR while compiling passage.0.1.0 ======================================#
# context              2.2.0~rc1 | linux/x86_64 | ocaml-base-compiler.5.0.0 | pinned(https://github.com/ahrefs/passage/releases/download/0.1.0/passage-0.1.0.tbz)
# path                 ~/.opam/5.0/.opam-switch/build/passage.0.1.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p passage -j 39 @install
# exit-code            1
# env-file             ~/.opam/log/passage-7-5485e0.env
# output-file          ~/.opam/log/passage-7-5485e0.out
### output ###
# (cd _build/default && /home/opam/.opam/5.0/bin/ocamlc.opt -open Stdlib -strict-formats -strict-sequence -short-paths -w +a-4@8-40..42-44-45-48-58-66-68-70-102 -g -bin-annot -I lib/.passage.objs/byte -I /home/opam/.opam/5.0/lib/astring -I /home/opam/.opam/5.0/lib/base -I /home/opam/.opam/5.0/lib/base/base_internalhash_types -I /home/opam/.opam/5.0/lib/base/caml -I /home/opam/.opam/5.0/lib/base/md5 -I /home/opam/.opam/5.0/lib/base/shadow_stdlib -I /home/opam/.opam/5.0/lib/base_bigstring -I /home/opam/.opam/5.0/lib/base_quickcheck -I /home/opam/.opam/5.0/lib/base_quickcheck/ppx_quickcheck/runtime -I /home/opam/.opam/5.0/lib/bin_prot -I /home/opam/.opam/5.0/lib/bin_prot/shape -I /home/opam/.opam/5.0/lib/biniou -I /home/opam/.opam/5.0/lib/bytes -I /home/opam/.opam/5.0/lib/camlp-streams -I /home/opam/.opam/5.0/lib/core -I /home/opam/.opam/5.0/lib/core/base_for_tests -I /home/opam/.opam/5.0/lib/core/validate -I /home/opam/.opam/5.0/lib/core_kernel/rope -I /home/opam/.opam/5.0/lib/curl -I /home/opam/.opam/5.0/lib/devkit -I /home/opam/.opam/5.0/lib/devkit/core -I /home/opam/.opam/5.0/lib/devkit/ocamlnet_lite -I /home/opam/.opam/5.0/lib/easy-format -I /home/opam/.opam/5.0/lib/extlib -I /home/opam/.opam/5.0/lib/extunix -I /home/opam/.opam/5.0/lib/fieldslib -I /home/opam/.opam/5.0/lib/fileutils -I /home/opam/.opam/5.0/lib/fpath -I /home/opam/.opam/5.0/lib/gen -I /home/opam/.opam/5.0/lib/int_repr -I /home/opam/.opam/5.0/lib/jane-street-headers -I /home/opam/.opam/5.0/lib/libevent -I /home/opam/.opam/5.0/lib/lwt -I /home/opam/.opam/5.0/lib/lwt/unix -I /home/opam/.opam/5.0/lib/menhirLib -I /home/opam/.opam/5.0/lib/ocaml/str -I /home/opam/.opam/5.0/lib/ocaml/threads -I /home/opam/.opam/5.0/lib/ocaml/unix -I /home/opam/.opam/5.0/lib/ocplib-endian -I /home/opam/.opam/5.0/lib/ocplib-endian/bigstring -I /home/opam/.opam/5.0/lib/parsexp -I /home/opam/.opam/5.0/lib/pcre -I /home/opam/.opam/5.0/lib/ppx_assert/runtime-lib -I /home/opam/.opam/5.0/lib/ppx_bench/runtime-lib -I /home/opam/.opam/5.0/lib/ppx_compare/runtime-lib -I /home/opam/.opam/5.0/lib/ppx_enumerate/runtime-lib -I /home/opam/.opam/5.0/lib/ppx_expect/collector -I /home/opam/.opam/5.0/lib/ppx_expect/common -I /home/opam/.opam/5.0/lib/ppx_expect/config -I /home/opam/.opam/5.0/lib/ppx_expect/config_types -I /home/opam/.opam/5.0/lib/ppx_hash/runtime-lib -I /home/opam/.opam/5.0/lib/ppx_here/runtime-lib -I /home/opam/.opam/5.0/lib/ppx_inline_test/config -I /home/opam/.opam/5.0/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/5.0/lib/ppx_log/types -I /home/opam/.opam/5.0/lib/ppx_module_timer/runtime -I /home/opam/.opam/5.0/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/5.0/lib/re2 -I /home/opam/.opam/5.0/lib/re2/c -I /home/opam/.opam/5.0/lib/sedlex -I /home/opam/.opam/5.0/lib/seq -I /home/opam/.opam/5.0/lib/sexplib -I /home/opam/.opam/5.0/lib/sexplib0 -I /home/opam/.opam/5.0/lib/splittable_random -I /home/opam/.opam/5.0/lib/stdio -I /home/opam/.opam/5.0/lib/stdlib-shims -I /home/opam/.opam/5.0/lib/time_now -I /home/opam/.opam/5.0/lib/trace/core -I /home/opam/.opam/5.0/lib/typerep -I /home/opam/.opam/5.0/lib/variantslib -I /home/opam/.opam/5.0/lib/yojson -I /home/opam/.opam/5.0/lib/zip -no-alias-deps -open Passage -o lib/.passage.objs/byte/passage__Template_parser.cmi -c -intf lib/template_parser.pp.mli)
# File "lib/template_parser.mli", line 15, characters 60-63:
# 15 | val template: (Lexing.lexbuf -> token) -> Lexing.lexbuf -> (ast)
#                                                                  ^^^
# Error: Unbound type constructor ast
```

File: packages/passage/passage.0.1.0/opam [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  ["dune" "subst"] {dev}
+  [
+    "dune"
+    "build"
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
```
Comment (mseri) [inline packages/passage/passage.0.1.0/opam:36]:
```suggestion
#    "@runtest" {with-test}  # require network access, blocked by the sandbox
```

**Comment by mseri** [comment]
There is still an incorrect/missing lower-bound, can you have a look? [lower-bounds (failed: Unbound type constructor ast)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/2e0e8a44344f83bc071b724a2f5eaf8da5c97477/variant/compilers,4.14,passage.0.1.0,lower-bounds)

**Comment by thatportugueseguy** [comment]
Yes. This compiled version of that `.mly` file is different from what we get on the same OS and ocaml version. I need to figure out what the issue is. thanks for looking into it too

**Comment by thatportugueseguy** [comment]
Added changes to fix the issue and opened a new PR.

Closing this PR in favor of https://github.com/ocaml/opam-repository/pull/26234.

**Pull request closed without merging**